lib/libc/sys/poll.c

```diff
@@ -0,0 +1,74 @@
+#include <sys/poll.h>
+#include <sys/time.h>
+#include <sys/event.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+
+int poll(struct pollfd *fds, nfds_t nfds, int timeout) {
+  int kq;
+  int ret;
+  struct kevent *events;
+  struct timespec timeout_ts;
+  int nevents = 0;
+
+  if (nfds > (unsigned int)sysconf(_SC_OPEN_MAX)) {
+    errno = EINVAL;
+    return -1;
+  }
+
+  if (timeout >= 0) {
+    timeout_ts.tv_sec = timeout / 1000;
+    timeout_ts.tv_nsec = (timeout % 1000) * 1000000;
+  }
+  /* if timeout == -1, then poll should wait indefinitely */
+  else if (timeout < -1) {
+    errno = EINVAL;
+    return -1;
+  }
+
+  kq = kqueue();
```

Comment:
It's not an issue now but if we will add proper threading support you can leak fd here.

Let's consider the following scenario:
```
td1:                             td2:
  - poll
    - kqueue
                      <leak>        - fork
    - close
```

*BSDs introduced `kqueue1`  as fix for that issue (no atomic close-on-exec). It allows to pass O_CLOEXEC as argument with standard semantic of this flag.

It's probably not a problem now but it will beat us in the future ;)

Reply:
Yes. We already have only `kqueue1` and `kqueue` just calls it with `0`.

I think that they should be inherited by a child - but it's a different discussion that is out of scope.

Reply:
Added O_CLOEXEC in https://github.com/cahirwpz/mimiker/pull/1415/commits/9eb99046af9a71baf4e6a6c1b0221bb93f063b6d